Add @nebular/theme Error

<!--
We will close this issue if you don't provide the needed information.

Please remember, the github issues is __NOT__ for support requests and general questions. It is for bugs and feature requests only.
Please read https://github.com/akveo/nebular/blob/master/CONTRIBUTING.md and search
existing issues (both open and closed) prior to opening any new issue and ensure you follow the instructions therein.
-->

### Issue type

**I'm submitting a ...**  (check one with "x")

* [x] bug report
* [ ] feature request

### Issue description

**Current behavior:**
Error install nebular 

**No matching version found for @schematics/angular@^8.1.3**

<!-- Describe how the bug manifests. -->
```
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @schematics/angular@^8.1.3
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget 
npm ERR! notarget It was specified as a dependency of 'test-app'
npm ERR! notarget 

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/user/.npm/_logs/2019-07-27T21_07_25_185Z-debug.log
```
**Expected behavior:**
Install Success
<!-- Describe what the behavior would be without the bug. -->

**Steps to reproduce:**
<!--  Please explain the steps required to duplicate the issue, especially if you are able to provide a sample application. -->
```
$ npm add @nebular/theme 
```



### Other information:

**npm, node, OS, Browser**
```


   / \   _ __   __ _ _   _| | __ _ _ __     / ___| |   |_ _|
   / △ \ | '_ \ / _` | | | | |/ _` | '__|   | |   | |    | |
  / ___ \| | | | (_| | |_| | | (_| | |      | |___| |___ | |
 /_/   \_\_| |_|\__, |\__,_|_|\__,_|_|       \____|_____|___|
                |___/
    

Angular CLI: 8.1.2
Node: 10.15.3
OS: linux x64
Angular: 8.1.3
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.801.2
@angular-devkit/build-angular      0.801.2
@angular-devkit/build-ng-packagr   0.801.2
@angular-devkit/build-optimizer    0.801.2
@angular-devkit/build-webpack      0.801.2
@angular-devkit/core               8.1.2
@angular-devkit/schematics         8.1.2
@angular/cdk                       8.1.1
@angular/cli                       8.1.2
@ngtools/json-schema               1.1.0
@ngtools/webpack                   8.1.2
@schematics/angular                8.1.2
@schematics/update                 0.801.2
ng-packagr                         5.3.0
rxjs                               6.5.2
typescript                         3.4.5
webpack                            4.35.2
```

Same with Angular CLI 8.3.1 because the `ng-add` schematics is using the `angular/core` version which does not exist on npm, you can check the Versions tab [here](https://www.npmjs.com/package/@schematics/angular)
The reason is that the CLI repo, Material repo and the Angular framework repo are not in sync only the major versions are of the Angular platform are in sync. This #1937 will fix it.

```
Installed packages for tooling via npm.
UPDATE package.json (1040 bytes)
npm ERR! code ETARGET
npm ERR! notarget No matching version found for @schematics/angular@~8.2.3
npm ERR! notarget In most cases you or one of your dependencies are requesting
npm ERR! notarget a package version that doesn't exist.
npm ERR! notarget
npm ERR! notarget It was specified as a dependency of 'angular-v831'
npm ERR! notarget
```

```
Angular CLI: 8.3.1
Node: 10.13.0
OS: win32 x64
Angular: 8.2.3
... animations, common, compiler, compiler-cli, core, forms
... language-service, platform-browser, platform-browser-dynamic
... router
```

--- a/src/framework/theme/schematics/ng-add/index.ts
+++ b/src/framework/theme/schematics/ng-add/index.ts
@@ -17,6 +17,7 @@ import {
   addDependencyToPackageJson,
   addDevDependencyToPackageJson,
   getDependencyVersionFromPackageJson,
+  getDevDependencyVersionFromPackageJson,
   getNebularPeerDependencyVersionFromPackageJson,
   getNebularVersion,
 } from '../util/package';
@@ -31,8 +32,9 @@ export default function (options: Schema): Rule {
 /**
  * Add required peer dependencies in package.json
  * */
-function installMainPeerDependencies(tree) {
+function installMainPeerDependencies(tree: Tree) {
   const angularCoreVersion = getDependencyVersionFromPackageJson(tree, '@angular/core');
+  const angularCliVersion = getDevDependencyVersionFromPackageJson(tree, '@angular/cli');
   const nebularThemeVersion = getNebularVersion();
   const angularCdkVersion = getNebularPeerDependencyVersionFromPackageJson('@angular/cdk');
 
@@ -41,7 +43,7 @@ function installMainPeerDependencies(tree) {
   addDependencyToPackageJson(tree, '@nebular/theme', nebularThemeVersion);
   addDependencyToPackageJson(tree, '@nebular/eva-icons', nebularThemeVersion);
 
-  addDevDependencyToPackageJson(tree, '@schematics/angular', angularCoreVersion);
+  addDevDependencyToPackageJson(tree, '@schematics/angular', angularCliVersion);
 }
 
 /**

--- a/src/framework/theme/schematics/util/package.ts
+++ b/src/framework/theme/schematics/util/package.ts
@@ -93,7 +93,7 @@ export function getDevDependencyVersionFromPackageJson(tree: Tree, packageName: 
 
   const packageJson: PackageJson = readJSON(tree, packageJsonName);
 
-  if (noInfoAboutDependency(packageJson, packageName)) {
+  if (noInfoAboutDevDependency(packageJson, packageName)) {
     throwNoPackageInfoInPackageJson(packageName);
   }
 
@@ -135,6 +135,13 @@ function noInfoAboutDependency(packageJson: PackageJson, packageName: string): b
 }
 
 /**
+ * Validates packageJson has devDependencies, also as specified devDependency not exists.
+ * */
+function noInfoAboutDevDependency(packageJson: PackageJson, packageName: string): boolean {
+  return !devDependencyAlreadyExists(packageJson, packageName);
+}
+
+/**
  * Validates packageJson has peerDependencies, also as specified peerDependency not exists.
  * */
 function noInfoAboutPeerDependency(packageJson: PackageJson, packageName: string): boolean {
@@ -146,6 +153,13 @@ function noInfoAboutPeerDependency(packageJson: PackageJson, packageName: string
  * */
 function dependencyAlreadyExists(packageJson: PackageJson, packageName: string): boolean {
   return !!(packageJson.dependencies && packageJson.dependencies[packageName]);
+}
+
+/**
+ * Validates packageJson has devDependencies, also as specified devDependency exists.
+ * */
+function devDependencyAlreadyExists(packageJson: PackageJson, packageName: string): boolean {
+  return !!(packageJson.devDependencies && packageJson.devDependencies[packageName]);
 }
 
 /**

--- a/src/framework/theme/schematics/util/package.ts
+++ b/src/framework/theme/schematics/util/package.ts
@@ -83,6 +83,23 @@ export function addDependencyToPackageJson(tree: Tree, packageName: string, pack
   writeJSON(tree, packageJsonName, packageJson);
 }
 
+/**
+ * Gets the version of the specified dev dependency by looking at the package.json in the specified tree
+ * */
+export function getDevDependencyVersionFromPackageJson(tree: Tree, packageName: string): string {
+  if (!tree.exists(packageJsonName)) {
+    throwNoPackageJsonError();
+  }
+
+  const packageJson: PackageJson = readJSON(tree, packageJsonName);
+
+  if (noInfoAboutDependency(packageJson, packageName)) {
+    throwNoPackageInfoInPackageJson(packageName);
+  }
+
+  return packageJson.devDependencies[packageName];
+}
+
 export function addDevDependencyToPackageJson(tree: Tree, packageName: string, packageVersion: string) {
   if (!tree.exists(packageJsonName)) {
     throwNoPackageJsonError();

